### PR TITLE
Adds Schedule and Slot resources.

### PR DESCRIPTION
**Pull request by msegedi**

### CODE REVIEW

File: spark_auto_mapper_fhir/resources/schedule.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from spark_auto_mapper_fhir.resources.practitioner import Practitioner
+from spark_auto_mapper_fhir.resources.practitioner_role import PractitionerRole
+from spark_auto_mapper_fhir.resources.related_person import RelatedPerson
+
+
+class Schedule(FhirResourceBase):
+    # noinspection PyPep8Naming
+    def __init__(
+        self,
+        id_: FhirId,
```
Comment (imranq2) [inline spark_auto_mapper_fhir/resources/schedule.py:27]: I believe this field is optional.  Did you use the New->FhirResource template?  That would set these automatically.

File: spark_auto_mapper_fhir/resources/schedule.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        actor: FhirList[Reference[Union[Patient, Practitioner,
+                                        PractitionerRole, RelatedPerson,
+                                        Device, HealthcareService, Location]]],
+        meta: Optional[Meta] = None,
+        identifier: Optional[FhirList[Identifier]] = None,
+        extension: Optional[FhirList[ExtensionBase]] = None,
+        active: Optional[FhirBoolean] = None,
```
Comment (imranq2) [inline spark_auto_mapper_fhir/resources/schedule.py:34]: Usually we add all the properties of a resource when we work on it so you can add the rest too

File: spark_auto_mapper_fhir/resources/slot.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+from spark_auto_mapper_fhir.resources.schedule import Schedule
+from spark_auto_mapper_fhir.valuesets.service_type import ServiceTypeCode
+from spark_auto_mapper_fhir.valuesets.slot_status import SlotStatusCode
+
+
+class Slot(FhirResourceBase):
+    # noinspection PyPep8Naming
+    def __init__(
+        self,
+        id_: FhirId,
```
Comment (imranq2) [inline spark_auto_mapper_fhir/resources/slot.py:24]: same comment with id being optional

File: spark_auto_mapper_fhir/resources/slot.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        schedule: Reference[Schedule],
+        status: SlotStatusCode,
+        start: FhirInstant,
+        end: FhirInstant,
+        meta: Optional[Meta] = None,
+        identifier: Optional[FhirList[Identifier]] = None,
```
Comment (imranq2) [inline spark_auto_mapper_fhir/resources/slot.py:30]: just add the remaining properties too from FHIR for this resource

File: spark_auto_mapper_fhir/resources/slot.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+        :param id_: id of resource
+        :param schedule: The schedule resource that this slot defines an interval of status information
+        :param status: busy | free | busy-unavailable | busy-tentative | entered-in-error
+        :param start: Date/Time that the slot is to begin
+        :param end: Date/Time that the slot is to conclude
+        :param identifier: An identifier for the slot
+        :param serviceType: The type of appointments that can be booked into this slot (ideally this would be an identifiable service - which is at a location, rather than the location itself). If provided then this overrides the value provided on the availability resource
+        """
+        super().__init__(
```
Comment (imranq2) [inline spark_auto_mapper_fhir/resources/slot.py:49]: the order here should ,match the order in http://hl7.org/fhir/slot.html.  We apply the schema and Spark will complain if the order is different

**Review by imranq2 — APPROVED** [review]

File: spark_auto_mapper_fhir/resources/schedule.py (hunk 2 above)
Comment (msegedi) [reply spark_auto_mapper_fhir/resources/schedule.py:40]: Done!

File: spark_auto_mapper_fhir/resources/slot.py (hunk 5 above)
Comment (msegedi) [reply spark_auto_mapper_fhir/resources/slot.py:67]: Fixed.

**Pull request merged**